docs/contributing/committing.rst

```diff
@@ -42,6 +45,8 @@ You can fix up any small mistakes in the commits,
 such as typos and formatting, as part of the rebase.
 ``git rebase --interactive`` is an excellent tool for this job.
 
+Ideally, use this as an opportunity to squash the changes to a few commits, so each commit is making a single meaningful change (and not breaking anything). If this is not possible because of the nature of the changes, go with the least ugly.
```

Comment:
As mentioned at the team meeting, I think "least ugly" is best open to the committer's interpretation - I don't think there's a good way to formalise it further.

However, I think "least ugly" needs more explanation in this sentence, because we haven't specified what the alternatives are. Maybe: "If this is not possible because of the nature of the changes, it's acceptable to either squash into a commit or leave all commits unsquashed, depending on which will be more readable in the commit history."